### DISCUSSION TITLE
Add SLO functionality

### DISCUSSION THREAD

**Discussion by ghost** [upptime/upptime#815] [Ideas]
It would be a really rich feature to add on SLOs and I think there is a lack of options for this in the OSS space. All of the groundwork for tracking latency and uptime SLOs is already present, it would just require a little extra work to get them to display as overlays in the graphs and be configurable as separate alert conditions.

Some use cases:

1. I have a site that only has a 99% uptime SLA - I don't want to alert every time its down, only if I breach the 99% uptime for a given period (week, month, year, etc.)
2. I have a site where the latency is important and have an SLA for it - I want to alert when the latency crosses it, even if the site is still up (99% already done, just branded as "degraded" currently)
3. I have an uptime SLA and can engage in additional downtime activities (patching, deployments, etc.) at the end of the month if I have unused downtime left - being able to see my downtime budget remaining would be really helpful.

**Comment by AnandChowdhary** [comment]
This is a good idea!

**Reply by ghost** [reply]
@AnandChowdhary I can pitch in for this if you're game, but have yet to do a deep dive on the code so am not sure the best place to start or help. 

Is there a way you'd like to proceed with it? I feel like #2 would maybe be an easy entry point that I could make a PR for which would be:

* Renaming keywords 'degraded'/'target' to 'Latency SLO' (or similar)
* Adding the SLO value as an annotation to the historical charts

But whatever you think is best :)
